Index: kernel-source/drivers/net/wireless/rtl8723ds/Makefile
===================================================================
--- kernel-source.orig/drivers/net/wireless/rtl8723ds/Makefile
+++ kernel-source/drivers/net/wireless/rtl8723ds/Makefile
@@ -20,8 +20,8 @@ ifeq ($(GCC_VER_49),1)
 EXTRA_CFLAGS += -Wno-date-time	# Fix compile error && warning on gcc 4.9 and later
 endif
 
-EXTRA_CFLAGS += -I$(src)/include
-EXTRA_CFLAGS += -I$(src)/hal/phydm
+EXTRA_CFLAGS += -I$(srctree)/$(src)/include
+EXTRA_CFLAGS += -I$(srctree)/$(src)/hal/phydm
 
 EXTRA_LDFLAGS += --strip-debug
 
@@ -230,11 +230,11 @@ _OUTSRC_FILES := hal/phydm/phydm_debug.o
 		hal/phydm/phydm_ccx.o
 
 
-EXTRA_CFLAGS += -I$(src)/platform
+EXTRA_CFLAGS += -I$(srctree)/$(src)/platform
 _PLATFORM_FILES := platform/platform_ops.o
 
 ifeq ($(CONFIG_BT_COEXIST), y)
-EXTRA_CFLAGS += -I$(src)/hal/btc
+EXTRA_CFLAGS += -I$(srctree)/$(src)/hal/btc
 _OUTSRC_FILES += hal/btc/HalBtc8192e1Ant.o \
 				hal/btc/HalBtc8192e2Ant.o \
 				hal/btc/HalBtc8723b1Ant.o \
